### ISSUE TITLE
[Bug]: EnumGetValuesTrait getValues must be static

### ISSUE THREAD

**Issue by mattamon** [pimcore/pimcore#17701]
The method must be made static, because you cannot open it through the SelectOptions, because the SelectOptions are an enum, and you cannot create an instance of enums, which is why you can only access the method if you go through a case/value/enum value, but this becomes difficult if the enum does not contain a value because the SelectOptions are empty.

Was fixed here https://github.com/pimcore/pimcore/pull/17349 but needed to be reverted because of https://github.com/pimcore/pimcore/issues/17554

Interface needs to change too

**Comment by fashxp** [comment]
bc break then?

**Comment by mattamon** [comment]
> bc break then?

Yes due to the interface change it is a BC-Break as I mentioned here.  https://github.com/pimcore/pimcore/pull/17349#issuecomment-2325946783

**Comment by NiklasBr** [comment]
How do we fix this in v11? Except for not using Select Options at all that is? As soon as we create a new Select Option in 11.3.2 Pimcore crashes because of the class does not match the interface.

**Comment by NiklasBr** [comment]
How do we fix this in v11? Except for not using Select Options at all that is? As soon as we create a new Select Option in Enterprise v11.3.2.1 Pimcore crashes because of the class does not match the interface.

**Comment by mattamon** [comment]
> How do we fix this in v11? Except for not using Select Options at all that is? As soon as we create a new Select Option in Enterprise v11.3.2.1 Pimcore crashes because of the class does not match the interface.

Hey @NiklasBr , we discussed it again internally and like I mentioned in my comment here https://github.com/pimcore/pimcore/pull/17349#issuecomment-2325946783 if we are really strict and we fix it it is a BC.
On the otherhand with the enums as described in the original PR it could have never worked in the first place like this. 

We will provided a Bugfix PR for this to solve this problem.
